File: src/types/sync.ts
export interface SyncResult {
  files_synced: number;
  files_failed: SyncError[];
  bytes_synced: number;
  duration_ms: number;
}

export interface SyncError {
  path: string;
  error: string;
}

export interface SyncProgress {
  session_id: string;
  message: string;
  current: number;
  total: number;
}

export interface SyncStatusEvent {
  session_id: string;
  status: 'idle' | 'syncing' | 'synced' | 'error';
}

File: src-tauri/src/models/sync.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SyncResult {
    pub files_synced: usize,
    pub files_failed: Vec<SyncError>,
    pub bytes_synced: u64,
    pub duration_ms: u64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SyncError {
    pub path: String,
    pub error: String,
}
